Tolerate win line endings on windows module_util load

jhawkesworth opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
I hit this when converting my playbooks to run under ansible 2.5 syntax.

It allows custom modules to load without hitting the following error message:

"Could not find imported module support code for 'Ansible.ModuleUtils.Legacy\r'
<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->
No existing bug report
##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->
windows module_utils
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.6.0 (tolerate_win_line_endings_on_module_load ed1d4c9120) last updated 2018/03/12 08:14:22 (GMT +100)
  config file = None
  configured module search path = [u'/home/jon/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/jon/ansible-dev/lib/ansible
  executable location = /home/jon/ansible-dev/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->
test using:

ansible-test windows-integration win_module_utils
<!--- Paste verbatim command output below, e.g. before and after your change -->
```
before, 
"Could not find imported module support code for 'Ansible.ModuleUtils.Legacy\r'
after
new win_module_util tests complete.
```

Commit:
tolerate windows line endings when loading windows module utils.  Helpful for old custom windows modules.

Commit:
add test modules to demonstrate win line ending module load behaviour.

Commit:
attempt to fix sanity check failures

jhawkesworth commented:
I pushed some changes which I hope will fix up the sanity tests (I don't seem to be able to run sanity checks locally)

jhawkesworth commented:
Third time lucky, shebang check now passes for me as well.

jhawkesworth commented:
4th time lucky, perhaps

jhawkesworth commented:
Failure in shippable seems unrelated 
{
"changed": false, 
"msg": "Missing required github3 module. (check docs or install with: pip install github3.py==1.0.0a4)"
}

mattclay commented:
That failure is unrelated. I've disabled the affected test and restarted CI.

jhawkesworth commented:
ready_for_review

jhawkesworth commented:
I don't understand the failure in shippable.

Commit:
pep8 fix

Commit:
explict skip of test modules from shebang check (core modules must still have expected unix style line endings)

jhawkesworth commented:
Revised as discussed at WWG yesterday
ready_for_review

Commit:
switch to rstrip() following core team meeting feedback

jhawkesworth commented:
Following feedback from core team meeting 3/Apr/18 I have changed to using rstrip() instead of rstrip(\r\n) and inlined the rstrip call.  

Hoping this wll make 2.5.1 if tests pass.

jhawkesworth commented:
hmm, not sure if test failure is unrelated or not, closing and reopening to trigger CI again.

jborean93 commented:
@jhawkesworth looks like the errors are unrelated, the Windows and sanity tests pass which is what I really care about in this change.

jborean93 reviewed (approved):
Looks great, thanks @jhawkesworth

jborean93 commented:
CI failures are unrelated to the code changes.

jborean93 commented:
@jhawkesworth are you able to create a backport PR for stable-2.5 so this is fixed in subsequent 2.5 releases?

sivel commented:
Backport PR instructions can be found at http://docs.ansible.com/ansible/latest/community/development_process.html#backport-pull-request-process

changelog entry goes into `changelog/fragments` in stable-2.5

jhawkesworth commented:
@jborean93 @sivel Thanks for the merge and the backport PR instructions.
I tried to create backport PR but I get merge conflicts, I think because the code smell scripts are now all .py in devel instead of bash scripts in stable-2.5

Maybe its best not to backport?

```
jon@TENSY ~/ansible-dev (backport/2.5/37291) $ git cherry-pick -x ad94d03ba1a2771e0b7e88e2083a4507a2767862
error: could not apply ad94d03... Tolerate win line endings on windows module_util load (#37291)
hint: after resolving the conflicts, mark the corrected paths
hint: with 'git add <paths>' or 'git rm <paths>'
hint: and commit the result with 'git commit'
jon@TENSY ~/ansible-dev (backport/2.5/37291) $ git status
On branch backport/2.5/37291
Your branch is up-to-date with 'upstream/stable-2.5'.

You are currently cherry-picking commit ad94d03.
  (fix conflicts and run "git cherry-pick --continue")
  (use "git cherry-pick --abort" to cancel the cherry-pick operation)

Changes to be committed:

        modified:   lib/ansible/executor/module_common.py
        new file:   test/integration/targets/win_module_utils/library/legacy_only_new_way_win_line_ending.ps1
        new file:   test/integration/targets/win_module_utils/library/legacy_only_old_way_win_line_ending.ps1
        modified:   test/integration/targets/win_module_utils/tasks/main.yml

Unmerged paths:
  (use "git add/rm <file>..." as appropriate to mark resolution)

        deleted by us:      test/sanity/code-smell/line-endings.py
        deleted by us:      test/sanity/code-smell/shebang.py

```
